tests/modification-check.py

```diff
@@ -0,0 +1,122 @@
+#!/usr/bin/env python
+
+from __future__ import print_function
+
+import os
+import subprocess
+import sys
+import tempfile
+import shutil
+try:
+    from urllib.request import urlretrieve
+except ImportError:
+    from urllib import urlretrieve
+
+def find_repo_path():
+    return os.path.dirname(os.path.dirname(os.path.realpath(__file__)))
+
+# We do not use filecmp.cmp to not be sensitive to CRLF/LF during comparison
+def compare_files(path_1, path_2):
+    l1 = l2 = True
+    with open(path_1, 'rU') as f1, open(path_2, 'rU') as f2:
```

Comment:
nit: It appears that [open](https://docs.python.org/3/library/functions.html#open) uses "universal newline mode" by default in Python 3. If it has this behavior in Python 3 and we're not planning to support Windows with Python 2, can the `U` here be removed? I'm largely only interested because using `U` is deprecated in newer versions of Python.

Reply:
Yes you are right. But it means that you should never have Python 2.7 as the default environment in your `PATH` on Windows, otherwise it will fail.

What bother me is that the requirements we are talking about here are for the tests workflow, not Certbot itself. In theory on Linux, you can run `tox` on any Python version, and envs requiring a specific version (like testing Certbot) specifies this directly in `tox.ini`. Then it will not be the case on Windows.

But for the sake of not using a deprecated option, I can live with that ^^. I just put a comment on the function to explain this.

Reply:
That's true.

For people exactly following our [current developer instructions](https://certbot.eff.org/docs/contributing.html#running-a-local-copy-of-the-client), that shouldn't be a problem because we have people create and activate a virtual environment containing Certbot and developer tools like `tox`, but because we don't plan on Certbot supporting Python 2 on Windows, Windows developers will have to be using a Python 3 environment. People more familiar with Python development could expect they can just run `tox` they have installed elsewhere without activating the environment though.

With your argument, I don't have a strong opinion here. If you have a significant preference for leaving `U` here, feel free to add it back.

In general, if you disagree with a change we recommend, please push back! It's certainly possible you thought of something we didn't which was the case here.

Reply:
Not a disagreement, more a work to find a compromise for various pro/con seen by different people around a design decision ^^ I will not hesitate on that, for sure !

I think you are right. You code a piece of software for a Python version, you use it for everything that is related to it. We just need to be very clear on the developer documentation about Python 3 and Windows when it will be official.